File: Java/Platform/Basics/MonitorTest/Program1.java
class Program {

	public static void main(String[] args) throws Exception {
		var acc = new JointAccount();
		acc.deposit(5000);
		System.out.println("Account opened for Jack and Jill.");
		System.out.printf("Initial Balance = %d%n", acc.balance());
		System.out.println("Jack is withdrawing 3000...");
		if(acc.withdraw(3000) == false)
			System.out.println("Jack's withdraw operation failed!");
		System.out.println("Jill is withdrawing 4000...");
		if(acc.withdraw(4000) == false)
			System.out.println("Jill's withdraw operation failed!");
		System.out.printf("Final Balance   = %d%n", acc.balance());
	}

	static class JointAccount {
		
		private int balance;

		public int balance() { return balance; }

		public void deposit(int amount) {
			balance = Worker.doWork(amount, balance, 1);
		}

		public boolean withdraw(int amount) {
			boolean success = false;
			if(balance >= amount){
				balance = Worker.doWork(amount, balance, -1);
				success = true;
			}
			return success;
		}
	}
}

File: dotnet/Platform/Basics/MonitorTest/DemoApp/Program1.cs
var acc = new JointAccount();
acc.Deposit(5000);
Console.WriteLine("Account opened for Jack and Jill.");
Console.WriteLine("Initial Balance = {0}", acc.Balance);
Console.WriteLine("Jack is withdrawing 3000...");
if(acc.Withdraw(3000) == false)
    Console.WriteLine("Jack's withdraw operation failed!");
Console.WriteLine("Jill is withdrawing 4000...");
if(acc.Withdraw(4000) == false)
    Console.WriteLine("Jill's withdraw operation failed!");
Console.WriteLine("Final Balance   = {0}", acc.Balance);


class JointAccount
{
    public int Balance { get; set; }

    public void Deposit(int amount)
    {
        Balance = Worker.DoWork(amount, Balance, 1);
    }

    public bool Withdraw(int amount)
    {
        bool success = false;
        if(Balance >= amount)
        {
            Balance = Worker.DoWork(amount, Balance, -1);
            success = true;
        }
        return success;
    }
}
